### PR TITLE
Redirect socat

### PR DESCRIPTION
**Pull request by slp**
Redirect socat output to avoid breaking console apps.

### CODE REVIEW

**Comment by teohhanhui** [comment]
Is the inclusion of this commit intentional?

> Introduce a client-server model

**Comment by teohhanhui** [comment]
`git checkout origin/main -- Cargo.lock`?

File: crates/krun-guest/src/pulse.rs [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+use utils::{env::find_in_path, stdout::create_stdout_stderr};
 
 pub fn setup_pulse_proxy(run_path: PathBuf) -> Result<()> {
     if let Some(socat_path) = find_in_path("socat")? {
         let pulse_path = run_path.join("pulse");
         fs::create_dir(&pulse_path)
             .context("Failed to create `pulse` directory in `XDG_RUNTIME_DIR`")?;
         let pulse_path = pulse_path.join("native");
+
+        let (stdout, stderr) = create_stdout_stderr("socat")?;
```
Comment (teohhanhui) [inline crates/krun-guest/src/pulse.rs:13]: WDYT about just using [`piped`](https://doc.rust-lang.org/std/process/struct.Stdio.html#method.piped)? Or is this important for debuggability?

Comment (slp) [reply crates/krun-guest/src/pulse.rs:13]: I think it could be useful in case something goes wrong with the pulseaudio bridge.

**Comment by slp** [comment]
> `git checkout origin/main -- Cargo.lock`?

Ouch.

File: crates/utils/src/lib.rs [hunk 2]
```diff
@@ -1,2 +1,3 @@
 pub mod env;
 pub mod launch;
+pub mod stdout;
```
Comment (teohhanhui) [inline crates/utils/src/lib.rs:3]:
```suggestion
pub mod stdio;
```

**Comment by teohhanhui** [comment]
> `git checkout origin/main -- Cargo.lock`?

Oops... Actually, the `Cargo.lock` in `main` is incorrect. That's why every time we do `cargo clippy` or `cargo build` it'll change `Cargo.lock`...

**Comment by slp** [comment]
Addressed suggestions and rebased on the latest HEAD.

**Review by teohhanhui — APPROVED** [review]
LGTM

**Pull request merged**